asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py

```diff
@@ -0,0 +1,850 @@
+                exp_data, oe_mols = oe_load_exp_from_file(
+                    args.mols,
+                    "sdf",
+                    return_mols=True,
+                    smiles_as_title=args.smiles_as_title,
+                )
+                used_3d = True
+                logger.info("setting used_3d to True")
+            else:
+                logger.info("Using 2D representation from SDF file")
+                exp_data = oe_load_exp_from_file(
+                    args.mols, "sdf", smiles_as_title=args.smiles_as_title
+                )
+
+        else:
+            raise ValueError(
+                f"Input molecules must be a SMILES file, SDF file, or SMILES string. Got {args.mols}"
+            )
+
+    logger.info(f"Loaded {len(exp_data)} molecules.")
+    if len(exp_data) == 0:
+        logger.error("No molecules loaded.")
+        raise ValueError("No molecules loaded.")
+
+    if args.verbose:
+        # we could make this just a debug statement but avoid looping over all molecules if not needed
+        for exp in exp_data:
+            logger.debug(f"Loaded molecule {exp.compound_id}: {exp.smiles}")
+
+    #########################
+    #     protein prep      #
+    #########################
+
+    # parse prep arguments
+    prep_dir = output_dir / "prep"
+    prep_dir.mkdir(parents=True, exist_ok=True)
+    intermediate_files.append(prep_dir)
+
+    logger.info(f"Prepping receptor in {prep_dir} at {datetime.now().isoformat()}")
+
+    # check inputs
+    receptor = Path(args.receptor)
+    if receptor.suffix != ".pdb":
+        raise ValueError(f"Receptor must be a PDB file. Got {args.receptor}")
+
+    if not receptor.exists():
+        raise ValueError(f"Receptor file does not exist: {args.receptor}")
+
+    if args.ref_prot is not None:
+        ref_prot = Path(args.ref_prot)
+        if ref_prot.suffix != ".pdb":
+            raise ValueError(
+                f"Reference protein must be a PDB file. Got {args.ref_prot}"
+            )
+
+    if args.loop_db is not None:
+        loop_db = Path(args.loop_db)
+        if not loop_db.exists():
+            raise ValueError(f"Loop database file does not exist: {args.loop_db}")
+
+    if args.seqres_yaml is not None:
+        seqres_yaml = Path(args.seqres_yaml)
+        # check it exists
+        if not seqres_yaml.exists():
+            raise ValueError(f"SEQRES yaml file does not exist: {args.seqres_yaml}")
+
+        # load it
+        logger.info(f"Using SEQRES from {args.seqres_yaml}")
+        with open(args.seqres_yaml) as f:
+            seqres_dict = yaml.safe_load(f)
+        seqres = seqres_dict["SEQRES"]
+    else:
+        seqres = None
+
+    # load the receptor
+    receptor_name = receptor.stem
+    xtal = CrystalCompoundData(
+        str_fn=args.receptor, smiles=None, output_name=str(receptor_name)
+    )
+
+    logger.info(f"Loaded receptor {receptor_name} from {receptor}")
+    oe_prep_function(
```

Comment:
As of #238 being merged the best way to run prep will become: `asapdiscovery.modelling.protein_prep_workflow`